crates/services/relayer/src/service/get_logs.rs

```diff
@@ -109,3 +109,18 @@ where
     }
     Ok(())
 }
+
+fn sort_events_by_log_index(events: Vec<Log>) -> anyhow::Result<Vec<Log>> {
+    let mut with_indexes = events
+        .into_iter()
+        .map(|e| {
+            let log_index = e
+                .log_index
+                .ok_or(anyhow::anyhow!("Log missing `log_index`: {e:?}"))?;
```

Comment:
The alternative here would be set to 0 if None.

I think I prefer the error, but there are still ways for this to be non-deterministic, e.g. if there are more than one log with the same index.

So, maybe we need to have additional checks that there are no duplicates? And do we want to accept batches with missing log indices?

If we don't want to worry about all the edge cases, then we could just do .unwrap_or(0) instead of .ok_or?

Reply:
Talked to the team about this offline. It sounds like we are okay with throwing an error here and not including the batch. In practice, this value should always be `Some`. 

In a similar vein, we can probably assume they will always be in order and all unique.

Reply:
Are all events coming through this sort helper `fn` guaranteed to be grouped by da_height?

Reply:
They get sorted into blocks after they are sorted by index. So it will maintain order once sorted by block.